x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table_header.tsx

```diff
@@ -151,20 +252,171 @@ const SearchField = ({
           fullWidth
         />
       </EuiFlexItem>
-      <CurrentPageOfTotal pageSize={pageSize} total={totalRulesCount} />
+      <CurrentPageOfTotal
+        pageSize={pageSize}
+        total={totalRulesCount}
+        selectedRules={selectedRules}
+        refetchRulesStates={refetchRulesStates}
+        setSelectAllRules={setSelectAllRules}
+        setSelectedRules={setSelectedRules}
+      />
     </div>
   );
 };
 
-const CurrentPageOfTotal = ({ pageSize, total }: RuleTableCount) => (
-  <EuiFlexItem grow={true}>
-    <EuiSpacer size="xl" />
-    <EuiText size="xs" textAlign="left" color="subdued" style={{ marginLeft: '8px' }}>
-      <FormattedMessage
-        id="xpack.csp.rules.rulesTable.showingPageOfTotalLabel"
-        defaultMessage="Showing {pageSize} of of {total, plural, one {# rule} other {# rules}}"
-        values={{ pageSize, total }}
-      />
-    </EuiText>
-  </EuiFlexItem>
-);
+const CurrentPageOfTotal = ({
+  pageSize,
+  total,
+  selectedRules,
+  refetchRulesStates,
+  setSelectAllRules,
+  setSelectedRules,
+}: RuleTableCount) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const onPopoverClick = () => {
+    setIsPopoverOpen((e) => !e);
+  };
+
+  const postRequestChangeRulesStatus = useChangeCspRuleState();
+  const closePopover = () => setIsPopoverOpen(false);
```

Comment:
this function looks redundant as its only used once

Reply:
the closePopover function is being used twice
one when clicking the Bulk Actions popOver
one after clicking any of the Bulk Actions options

Reply:
i still think its redundant since its hold basically no logic, but your call on that one

Reply:
ah kk fair enough, I'll just call it directly then